xapian-bindings/android/CROSS_COMPILE

```diff
@@ -0,0 +1,121 @@
+export PATH=$PATH:$TOOLCHAIN/bin
+export OUTPUT_DIR=<path_to_xapian>/xapian-bindings/android/built
+cd <path_to_xapian>/xapian-core
+./configure --prefix=$OUTPUT_DIR --host=<host>  LDFLAGS="-fPIE -pie" --disable-shared --with-pic 
+make
+make install
+
+The above configure will build xapian statically with position independent code.
+
+Generating the bindings for android
+===================================
+Now generate the bindings for android as follows:
+
+cd <path_to_xapian>/xapian-bindings/android
+mkdir -p org/xapian
+swig -Werror -I. -I../../xapian-core/include -c++ -java -module Xapian -package org.xapian -outdir org/xapian -o xapian_wrap.cc android.i
+
+This should generate xapian_wrap.cc,xapian_wrap.h and .java files in org/xapian. Now add the following line in org/xapian/XapianJNI.java at start of static block at line no 1114:
+
+System.loadLibrary("xapian_jni");
+
+Save the file. Now generate the libxapian_jni.so by doing the following:
+
+export CXX=$TOOLCHAIN/bin/<host>-g++
+$CXX -c xapian_wrap.cc -I../../xapian-core/include
+$CXX -shared -o built/lib/libxapian_jni.so -Wl,-whole-archive libxapian-x.x.a  -Wl,-no-whole-archive xapian_wrap.o -lz -luuid -lc
+
+This will generate libxapian_jni.so in built/lib/libxapian_jni.so. Now compile all the .java class using :
+
+javac -source 1.7 -target 1.7 -bootclasspath <path_to_jvm>/<your_jdk>/jre/lib/rt.jar org/xapian/*.java
+
+<path_to_jvm> is the jvm installation path eg.usr/lib on Ubuntu
+The 'source' and 'target' flags are required for android compactibility if you have java 1.8 or above. 
+
+
+Now create a proper file structure for .aar for android by:
+
+mkdir -p aar/jni/<proper_architecture> aar/res
+jar -cf aar/classes.jar org/xapian/*.class
+cp built/lib/libxapian_jni.so aar/jni/<proper_architecture>/
+cd aar
+jar -cf ../built/xapianlib.aar *
+
+Now xapianlib.aar as a library contains libxapian_jni.so and jni bindings ready for use on android devices.
+
+Testing Xapian on android
+=========================
+1)Now to test your cross compiled build(libxapian-x.x.a) do the following:
+
+cd <path_to_xapian>/xapian-core/tests
+make apitest
+
+This will generate apitest file in tests folder. You will need to push apitest file, testdata folder and runtest script to your android device or emulator.
+
+To push a file to your android device use the following command:
+
+adb push <path_of_file> <dst_on_device>
+
+<dst_on_device> is the destination where you have the read-write-execute permission. eg. /data/local/tmp
+
+After pushing all the files to the device, you can run test as follows:
+
+adb shell
+cd <dst_on_device>
+sh runtest ./apitest
+
+2)Now to test your bindings, set the location of android sdk and android ndk in example/local.properties file and copy the xapianlib.aar in example/app/libs folder. To build the example and install the app on device or emulator run the following:
+
+cd example
+./gradlew build
+./gradlew installDebug
```

Comment:
I'm looking at this step now - it seems there's binary code in `gradle/wrapper/gradle-wrapper.jar` without any corresponding source code, so I'm not at all keen to run the commands above - I don't trust binary code I can't rebuild from source for myself if I want to.  We also don't want to be adding jar files to xapian git.

The Debian `libgradle-core-java` package contains `/usr/share/java/gradle-wrapper.jar` - is there a way to use a system version like that instead of bundling jar files?